### ISSUE TITLE
Removing a team from terraform causes an error due to attachment to project

### ISSUE THREAD

**Issue by checkitsam** [mongodb/terraform-provider-mongodbatlas#662]
Unable to delete team and remove from project in single operation, perhaps a separate attachment resource would help rather than having to define within the project resource. e.g. `mongodbatlas_teams_attachment` to allow the team to be attached to a project. 

### Terraform CLI and Terraform MongoDB Atlas Provider Version
```hcl
Terraform v1.1.4
on darwin_amd64
+ provider registry.terraform.io/mongodb/mongodbatlas v1.2.0
```

### Terraform Configuration File 
```hcl
locals {
  org_id = "ORG_ID_HERE"
  teams = {
    "Team1" = {
      usernames = ["my.user1@mycompany.com"]
    }
    "Team2" = {
        usernames  = ["my.user1@mycompany.com"]
    }
  }
}

resource "mongodbatlas_teams" "team" {
  for_each  = local.teams
  org_id    = local.org_id
  name      = each.key
  usernames = each.value.usernames
}

resource "mongodbatlas_project" "project" {
  name   = "Test-Project"
  org_id = local.org_id

  dynamic "teams" {
    for_each = mongodbatlas_teams.team
    content {
      role_names = ["GROUP_OWNER"]
      team_id    = teams.value.team_id
    }
  }
}
```

### Steps to Reproduce
1. terraform init
2. terraform apply
3. Remove a team from local.teams
4. terraform apply

### Expected Behavior
Team should be removed from project and then deleted

### Actual Behavior
Terraform attempts to delete the team first and errors due to its attachment to the project

### Debug Output 
```
Plan: 0 to add, 1 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

mongodbatlas_teams.team["Team1"]: Destroying... [id=]
╷
│ Error: error deleting Team (<team_id>): DELETE https://cloud.mongodb.com/api/atlas/v1.0/orgs/<org_id>/teams/<team_id>: 404 (request "CANNOT_DELETE_TEAM_ASSIGNED_TO_PROJECT") You cannot delete a team that has at least one project assigned to it. Make sure to remove the team from all project before deleting it.
```

**Comment by nikhil-mongo** [comment]
@checkitsam I could repro this and understand what is missing in terms of following the process. I will file the internal ticket to get this fixed.

**Comment by nikhil-mongo** [comment]
INTMDB-298 filed internally.

**Comment by themantissa** [comment]
The fix for this was just released in version 1.3.0.   Thanks!
